core/client/app/services/feature.js

```diff
@@ -0,0 +1,59 @@
+import Ember from 'ember';
+
+const {Service, computed, inject, PromiseProxyMixin, set} = Ember;
+
+export function feature(name) {
+    return computed(`config.${name}`, `labs.${name}`, {
+        get() {
+            return this.get(`config.${name}`) || this.get(`labs.${name}`);
+        },
+        set(key, value) {
+            this.update(name, value);
+            return value;
+        }
+    });
+}
+
+export default Service.extend(PromiseProxyMixin, {
+    store: inject.service(),
+    config: inject.service(),
+    notifications: inject.service(),
+
+    publicAPI: feature('publicAPI'),
+
+    labs: computed('isSettled', 'content.labs', function () {
+        let value = {};
+
+        if (this.get('isFulfilled')) {
+            try {
+                value = JSON.parse(this.get('content.labs') || {});
+            } catch (err) {
+                value = {};
+            }
+        }
+
+        return value;
+    }),
+
+    update(key, value) {
+        let labs = this.get('labs');
+        let content = this.get('content');
+
+        set(labs, key, value);
+        content.set('labs', JSON.stringify(labs));
+        content.save().catch((errors) => {
```

Comment:
@kevinansfield there seems to be some issue here in testing => the settings PUT endpoint doesn't seem to be called with this. I'm not quite sure why. :confused: